packages/iframe-execution-environment-service/src/IframeExecutionEnvironmentService.ts

```diff
@@ -0,0 +1,337 @@
+import { Duplex } from 'stream';
+import { nanoid } from 'nanoid';
+import pump from 'pump';
+import { ObservableStore } from '@metamask/obs-store';
+import ObjectMultiplex from '@metamask/object-multiplex';
+import { WindowPostMessageStream } from '@metamask/post-message-stream';
+import { PLUGIN_STREAM_NAMES } from '@mm-snap/workers';
+import { createStreamMiddleware } from 'json-rpc-middleware-stream';
+import { PluginData } from '@mm-snap/types';
+import {
+  JsonRpcEngine,
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+} from 'json-rpc-engine';
+import { ExecutionEnvironmentService } from '@mm-snap/controllers';
+
+export type SetupPluginProvider = (pluginName: string, stream: Duplex) => void;
+
+interface JobControllerArgs {
+  setupPluginProvider: SetupPluginProvider;
+  iframeUrl: URL;
+}
```

Comment:
Although this class has some similarities with controllers, it's not one in the sense that we use it, and we should name this `IframeExecutionServiceArgs` or something like that instead.